vespalib/src/vespa/vespalib/util/simple_thread_bundle.h

```diff
@@ -103,9 +103,21 @@ class SimpleThreadBundle : public ThreadBundle
         std::vector<SimpleThreadBundle*> _bundles;
 
     public:
+        class Guard {
```

Comment:
where is this used/tested?

Reply:
Used in matchengine.cpp
Added explicit test to this PR